### PR TITLE
Improve enable public file viewing

### PR DESCRIPTION
**Pull request by nick-benoit14**
We should only make the finally selected file publicly viewable.

### CODE REVIEW

File: lms/templates/content_item_selection/new_content_item_selection.html.jinja2 [hunk 1]
```diff
@@ -230,8 +253,10 @@
     function pickerCallback(data) {
       var docCount = data.docs && data.docs.length;
       if (data.action == google.picker.Action.PICKED && docCount === 1) {
```
Comment (keithrichardsaj) [inline lms/templates/content_item_selection/new_content_item_selection.html.jinja2]: Is it possible for more than 1 document to get picked? If so we should return an error to the user to let them know they should only pick 1 document at a time.

File: lms/templates/content_item_selection/new_content_item_selection.html.jinja2 [hunk 2]
```diff
@@ -230,8 +253,10 @@
...
-        enablePublicViewing(data.docs[0]);
-        var url = buildDocUrl(data.docs[0]);
+        var doc = data.docs[0];
+        var url = buildDocUrl(doc);
+        state.selectedDocId = doc.id;
+        state.selectedDocUrl = url;
         if(url) {
```
Comment (keithrichardsaj) [inline lms/templates/content_item_selection/new_content_item_selection.html.jinja2]: Should we have error handling if there is no url, or do you think the error handling in the function buildDocUrl is enough?

File: lms/templates/content_item_selection/new_content_item_selection.html.jinja2 [hunk 3]
```diff
@@ -59,6 +68,30 @@
       ]
     }
     form.elements['content_items'].value = JSON.stringify(contentItem)
+
+    // If the user submits the google picker selected url, then we should
+    // make that file public. If they use the picker and then change the url,
+    // we should not make the selected file public.
+    if(state.selectedDocId && state.selectedDocUrl && document_url === state.selectedDocUrl) {
+      // If we need to make a file public, then we need to stop the event to we
```
Comment (keithrichardsaj) [inline lms/templates/content_item_selection/new_content_item_selection.html.jinja2]: "event to we" -> "event so we" ?

Comment (nick-benoit14) [reply lms/templates/content_item_selection/new_content_item_selection.html.jinja2]: Done

File: lms/templates/content_item_selection/new_content_item_selection.html.jinja2 (hunk 1 above)
Comment (nick-benoit14) [reply lms/templates/content_item_selection/new_content_item_selection.html.jinja2]: The picker is configured to only allow a single selection

File: lms/templates/content_item_selection/new_content_item_selection.html.jinja2 (hunk 2 above)
Comment (nick-benoit14) [reply lms/templates/content_item_selection/new_content_item_selection.html.jinja2]: Added

**Pull request merged**